crates/sui-core/src/node_sync/node_follower.rs

```diff
@@ -336,11 +336,12 @@ where
         while let Some(DigestsMessage { sync_arg, tx }) = receiver.recv().await {
             let state = self.clone();
             let limit = limit.clone();
-            tokio::spawn(async move {
-                // hold semaphore permit until task completes. unwrap ok because we never close
-                // the semaphore in this context.
-                let permit = limit.acquire_owned().await.unwrap();
 
+            // hold semaphore permit until task completes. unwrap ok because we never close
+            // the semaphore in this context.
+            let permit = limit.acquire_owned().await.unwrap();
+
+            tokio::spawn(async move {
```

Comment:
Would it be ok to ask you to create a scope enclosing precisely the acquisition of this semaphore + the task it breaks?

As such, the semaphore only drops at the end of the function - will still be held if someone adds something at the end of this function. If that thing takes a while, that might be unintended.

Reply:
the permit is moved into the spawned task and then immediately passed to another function